utils/like_matcher.hh

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <boost/regex/icu.hpp>
+
+#include "bytes.hh"
+
+/// Implements <code>text LIKE pattern</code>.
+class like_matcher {
+public:
+    /// Compiles \c pattern and stores the result.
+    ///
+    /// @param pattern UTF-8 encoded pattern with wildcards '_' and '%'.
+    like_matcher(bytes_view pattern);
```

Comment:
`explicit`

Nit: we use the `\param` convention in our Doxygen comments. But Doxygen comments in Scylla are mostly inexistent and the existing few wildly inconsistent so its up to you to change this if you want.

Reply:
Done.